openedx/core/djangoapps/coursegraph/management/commands/tests/test_dump_to_neo4j.py

```diff
@@ -194,37 +244,42 @@ def test_coerce_types(self, original_value, coerced_expected):
         coerced_value = self.mss.coerce_types(original_value)
         self.assertEqual(coerced_value, coerced_expected)
 
-    def test_dump_to_neo4j(self):
+    @mock.patch('openedx.core.djangoapps.coursegraph.management.commands.dump_to_neo4j.NodeSelector')
+    def test_dump_to_neo4j(self, mock_selector_class):
         """
         Tests the dump_to_neo4j method works against a mock
         py2neo Graph
         """
-        mock_graph = mock.Mock()
-        mock_transaction = mock.Mock()
-        mock_graph.begin.return_value = mock_transaction
+        mock_graph = MockGraph()
+        mock_selector_class.return_value = MockNodeSelector(mock_graph)
 
         successful, unsuccessful = self.mss.dump_courses_to_neo4j(mock_graph)
 
-        self.assertEqual(mock_graph.begin.call_count, 2)
-        self.assertEqual(mock_transaction.commit.call_count, 2)
-        self.assertEqual(mock_transaction.rollback.call_count, 0)
+        self.assertCourseDump(
+            mock_graph,
+            number_of_courses=2,
+            number_commits=2,
+            number_rollbacks=0,
+        )
 
-        # 7 nodes + 9 relationships from the first course
+        # 9 nodes + 7 relationships from the first course
         # 2 nodes and no relationships from the second
-        self.assertEqual(mock_transaction.create.call_count, 18)
-        self.assertEqual(mock_transaction.run.call_count, 2)
+
+        self.assertEqual(len(mock_graph.nodes), 11)
 
         self.assertEqual(len(unsuccessful), 0)
         self.assertItemsEqual(successful, self.course_strings)
 
-    def test_dump_to_neo4j_rollback(self):
+    @mock.patch('openedx.core.djangoapps.coursegraph.management.commands.dump_to_neo4j.NodeSelector')
+    @mock.patch('openedx.core.djangoapps.coursegraph.management.commands.dump_to_neo4j.Graph')
+    def _test_dump_to_neo4j_rollback(self, mock_graph_class, mock_selector_class):
         """
         Tests that the the dump_to_neo4j method handles the case where there's
         an exception trying to write to the neo4j database.
         """
-        mock_graph = mock.Mock()
-        mock_transaction = mock.Mock()
-        mock_graph.begin.return_value = mock_transaction
+        mock_graph, mock_transaction = self.setup_mock_graph(
```

Comment:
how `self. setup_mock_graph` is returning multiple values? As I can see its only returning one. All the other places only one variable is used except here.